src/DataStructures/DataBox/DataBox.hpp

```diff
@@ -1605,13 +1619,56 @@ struct Apply<tmpl::list<Tags...>> {
  * \param args the arguments to pass to the function that are not in the
  * DataBox, `box`
  */
-template <typename TagsList, typename F, typename BoxTags, typename... Args>
+template <typename TagsList, typename F, typename BoxTags, typename... Args,
+          Requires<not DataBox_detail::has_argument_tags_v<std::decay_t<F>>> =
+              nullptr>
 inline constexpr auto apply(F&& f, const DataBox<BoxTags>& box,
-                            Args&&... args) {
+                            Args&&... args) noexcept {
   return DataBox_detail::Apply<TagsList>::apply(std::forward<F>(f), box,
                                                 std::forward<Args>(args)...);
 }
 
+template <
+    typename F, typename BoxTags, typename... Args,
+    Requires<DataBox_detail::has_argument_tags_v<std::decay_t<F>>> = nullptr,
+    typename ArgumentTags = typename std::decay_t<F>::argument_tags>
+inline constexpr auto apply(F&& f, const DataBox<BoxTags>& box,
```

Comment:
The `inline`s your adding should either be not added or `FORCE_INLINE`s

Reply:
I copied the `inline` from the other functions in this file. Should they be changed as well, and if so, how?

Reply:
I'm fine with just changing it in the code you actually wrote, and also with changing it elsewhere. Whatever you prefer :)